include/RAJA/pattern/teams/teams_core.hpp

```diff
@@ -0,0 +1,280 @@
+/*!
+ ******************************************************************************
+ *
+ * \file
+ *
+ * \brief   RAJA header file containing user interface for RAJA::Teams
+ *
+ ******************************************************************************
+ */
+
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2016-20, Lawrence Livermore National Security, LLC
+// and RAJA project contributors. See the RAJA/COPYRIGHT file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+#ifndef RAJA_pattern_teams_core_HPP
+#define RAJA_pattern_teams_core_HPP
+
+#include "RAJA/config.hpp"
+#include "RAJA/internal/get_platform.hpp"
+#include "RAJA/util/StaticLayout.hpp"
+#include "RAJA/util/macros.hpp"
+#include "RAJA/util/plugins.hpp"
+#include "RAJA/util/types.hpp"
+#include "camp/camp.hpp"
+#include "camp/concepts.hpp"
+#include "camp/tuple.hpp"
+
+#if defined(RAJA_ENABLE_CUDA) || defined(RAJA_ENABLE_HIP)
+#define RAJA_ENABLE_DEVICE
+#endif
+
+#if defined(RAJA_DEVICE_CODE)
+#define TEAM_SHARED __shared__
+#else
+#define TEAM_SHARED
+#endif
+
+namespace RAJA
+{
+
+namespace expt
+{
+
+// GPU or CPU threads available
+enum ExecPlace {
+  HOST
+#if defined(RAJA_ENABLE_DEVICE)
+  ,
+  DEVICE
+#endif
```

Comment:
Should the values of this enum vary when compiling with and without devices.

Reply:
I vote that it shouldn't vary

Reply:
What should happen when a device is not available? Should we just throw an error in that case?

Reply:
Fixed, it now throws an error when a user tries to use a device but it is not enabled.